torch/distributed/rpc/api.py

```diff
@@ -382,3 +383,98 @@ def rpc_async(to, func, args=None, kwargs=None):
     """
     fut = _invoke_rpc(to, func, args, kwargs)
     return fut
+
+
+# All below private APIs are for making rpc torch script call.
+# Right now, rpc torch script call requires qualified_name to look up
+# compliation unit function, so rpc torch script call APIs are different
+# from rpc APIs for python call and built in operator call.
+# These private APIs are subject to be merged with above public rpc APIs when
+# jit supports torch script funciton to be a jit type.
+@_require_initialized
+def _rpc_sync(to, qualified_name, args=None, kwargs=None):
+    r"""
+    Make a blocking RPC call to run function ``func`` on worker ``to``. RPC
+    messages are sent and received in parallel to execution of Python code. This
+    method is thread-safe.
+
+    Arguments:
+        to (str): name of the destination worker.
+        qualified_name (str): python function annotated with @torch.jit.script
+                              (like ``moduleName::torchScriptFuncName``)
+                              can be sent over RPC more efficiently.
+        args (tuple): the argument tuple for the ``func`` invocation.
+        kwargs (dict): is a dictionary of keyword arguments for the ``func``
+                       invocation.
+
+    Returns:
+        Returns the result of running ``func``on ``args`` and ``kwargs``.
+
+    Example::
+        On worker 0:
+        >>> import torch.distributed as dist
+        >>> import torch.distributed.rpc as rpc
+        >>> dist.init_process_group(backend='gloo', rank=0, world_size=2)
+        >>> rpc.init_model_parallel("worker0")
+        >>> ret = rpc.rpc_sync("worker1", "myModule::myTorchScriptFunc", args=(torch.ones(2), 3))
+        >>> rpc.join_rpc()
+
+        On worker 1:
+        >>> import torch.distributed as dist
+        >>> import torch.distributed.rpc as rpc
+        >>> dist.init_process_group(backend='gloo', rank=1, world_size=2)
+        >>> rpc.init_model_parallel("worker1")
+        >>> rpc.join_rpc()
+    """
+    args = args if args else ()
+    kwargs = kwargs if kwargs else {}
+    fut = _invoke_rpc_script(to, qualified_name, *args, **kwargs)
```

Comment:
I was wondering if we can avoid having a separate API for torchscript call. Is it possible to do this in our current RPC API?

```
try:
   qname = torch.jit._qualified_name(func)
   return _invoke_rpc_script(...)
except:
   return _invoke_rpc(...)
```